File: rb/features.rb
# OpenaqPlatform SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module OpenaqPlatformFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpenaqPlatformBaseFeature.new
    when "test"
      OpenaqPlatformTestFeature.new
    else
      OpenaqPlatformBaseFeature.new
    end
  end
end

File: py/features.py
# OpenaqPlatform SDK feature factory

from feature.base_feature import OpenaqPlatformBaseFeature
from feature.test_feature import OpenaqPlatformTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenaqPlatformBaseFeature(),
        "test": lambda: OpenaqPlatformTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
